### DISCUSSION TITLE
[v4] How can I check if user exists for a few routes with a middleware and not always manually?

### DISCUSSION THREAD

**Discussion by the-prodigy** [adonisjs/core#1725] [Help] [answered]
The question is above.

How can I create a middleware for like a user not found check or whats best practices to do that automatically for a few routes before reaching the controller or yeah

**Comment by thetutlage** [comment, marked as answer]
- Create a middleware
- Register it inside the `named` object in `start/kernel.ts` file.
- Create a group of routes on which you want to apply this middleware

```ts
Route.group(() => {
   // define routes on which the middleware should apply
}).middleware('registered-name-inside-kernel-file')
```

**Reply by the-prodigy** [reply]
Thanks! Helped!
